### ISSUE TITLE
JSON.sh does crash when parsing numerous nested structures

### ISSUE THREAD

**Issue by nst** [dominictarr/JSON.sh#47]
**Step to reproduce**

    $ python -c "print('['*100000)" | ./JSON.sh 
    
**Expected behaviour**

JSON.sh should either parse the string, or return an error.

**Actual behaviour**

    ./JSON.sh: line 206: 40694 Done                    tokenize
    40695 Segmentation fault: 11  | parse

JSON.sh does crash because it lacks a nesting limit.

**Comment by dominictarr** [comment]
I don't think we could have a nesting limit, other than exiting when it was reached, because even if you wanted to scan and not parse things beneath the limit, you'd still need to make sure the brackets balanced.
